make requirements fails

Creating an environment from scratch and installing requirements with `make requirements` fails due to densepose missing pytorch as a build requirement build:

```shell
$ make create_environment
$ conda activate zamba
$ make requirements
conda install cudatoolkit=11.0.3 cudnn=8.0 -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/robert/anaconda3/envs/zamba

  added / updated specs:
    - cudatoolkit=11.0.3
    - cudnn=8.0


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    cudatoolkit-11.0.3         |       h15472ef_8       951.9 MB  conda-forge
    cudnn-8.0.5.39             |       ha5ca753_1       535.4 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        1.45 GB

The following NEW packages will be INSTALLED:

  cudatoolkit        conda-forge/linux-64::cudatoolkit-11.0.3-h15472ef_8
  cudnn              conda-forge/linux-64::cudnn-8.0.5.39-ha5ca753_1
  python_abi         conda-forge/linux-64::python_abi-3.8-2_cp38

The following packages will be UPDATED:

  certifi            pkgs/main::certifi-2021.10.8-py38h06a~ --> conda-forge::certifi-2021.10.8-py38h578d9bd_1


Proceed ([y]/n)? y


Downloading and Extracting Packages
cudnn-8.0.5.39       | 535.4 MB  | #################################################################################################################################################################################################################################### | 100%
cudatoolkit-11.0.3   | 951.9 MB  | #################################################################################################################################################################################################################################### | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: | By downloading and using the CUDA Toolkit conda packages, you accept the terms and conditions of the CUDA End User License Agreement (EULA): https://docs.nvidia.com/cuda/eula/index.html

| By downloading and using the cuDNN conda packages, you accept the terms and conditions of the NVIDIA cuDNN EULA -
  https://docs.nvidia.com/deeplearning/cudnn/sla/index.html

done
python -m pip install -U pip setuptools wheel
Requirement already satisfied: pip in /home/robert/anaconda3/envs/zamba/lib/python3.8/site-packages (21.2.4)
Collecting pip
  Downloading pip-21.3.1-py3-none-any.whl (1.7 MB)
     |████████████████████████████████| 1.7 MB 2.6 MB/s
Requirement already satisfied: setuptools in /home/robert/anaconda3/envs/zamba/lib/python3.8/site-packages (58.0.4)
Collecting setuptools
  Downloading setuptools-58.5.3-py3-none-any.whl (946 kB)
     |████████████████████████████████| 946 kB 70.4 MB/s
Requirement already satisfied: wheel in /home/robert/anaconda3/envs/zamba/lib/python3.8/site-packages (0.37.0)
Installing collected packages: setuptools, pip
  Attempting uninstall: setuptools
    Found existing installation: setuptools 58.0.4
    Uninstalling setuptools-58.0.4:
      Successfully uninstalled setuptools-58.0.4
  Attempting uninstall: pip
    Found existing installation: pip 21.2.4
    Uninstalling pip-21.2.4:
      Successfully uninstalled pip-21.2.4
Successfully installed pip-21.3.1 setuptools-58.5.3
python -m pip install -r requirements-dev.txt
Obtaining file:///home/robert/projects/zamba (from -r requirements-dev.txt (line 1))
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting mike
  Downloading mike-1.1.2-py3-none-any.whl (26 kB)
Collecting mkdocs>=1.2.2
  Downloading mkdocs-1.2.3-py3-none-any.whl (6.4 MB)
     |████████████████████████████████| 6.4 MB 3.0 MB/s
Collecting mkdocs-material>=7.3.4
  Downloading mkdocs_material-7.3.6-py2.py3-none-any.whl (4.5 MB)
     |████████████████████████████████| 4.5 MB 48.4 MB/s
Collecting mkdocstrings>=0.15.2
  Downloading mkdocstrings-0.16.2-py3-none-any.whl (42 kB)
     |████████████████████████████████| 42 kB 529 kB/s
Collecting black
  Using cached black-21.10b0-py3-none-any.whl (150 kB)
Collecting flake8
  Using cached flake8-4.0.1-py2.py3-none-any.whl (64 kB)
Collecting pytorchvideo@ git+https://github.com/facebookresearch/pytorchvideo
  Cloning https://github.com/facebookresearch/pytorchvideo to /tmp/pip-install-p9zfm8hk/pytorchvideo_f4dad7c0eff140528d04757ebb1cea6a
  Running command git clone --filter=blob:none -q https://github.com/facebookresearch/pytorchvideo /tmp/pip-install-p9zfm8hk/pytorchvideo_f4dad7c0eff140528d04757ebb1cea6a
  Resolved https://github.com/facebookresearch/pytorchvideo to commit 168e16859a6029ef8ebeb476f9163bebb6c6b87d
  Preparing metadata (setup.py) ... done
Collecting yolox@ git+https://github.com/r-b-g-b/YOLOX.git@build-requirements
  Cloning https://github.com/r-b-g-b/YOLOX.git (to revision build-requirements) to /tmp/pip-install-p9zfm8hk/yolox_7969fbd19fa64256bf813835ed5f8a69
  Running command git clone --filter=blob:none -q https://github.com/r-b-g-b/YOLOX.git /tmp/pip-install-p9zfm8hk/yolox_7969fbd19fa64256bf813835ed5f8a69
  Running command git checkout -b build-requirements --track origin/build-requirements
  Switched to a new branch 'build-requirements'
  Branch 'build-requirements' set up to track remote branch 'build-requirements' from 'origin'.
  Resolved https://github.com/r-b-g-b/YOLOX.git to commit 3631ad70d1b01d5988ff5745655eb2ff182d5bd9
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting detectron2-densepose@ git+https://github.com/facebookresearch/detectron2@main#subdirectory=projects/DensePose
  Cloning https://github.com/facebookresearch/detectron2 (to revision main) to /tmp/pip-install-p9zfm8hk/detectron2-densepose_ab83aaa5a59c4ff4be17e4db5033ec05
  Running command git clone --filter=blob:none -q https://github.com/facebookresearch/detectron2 /tmp/pip-install-p9zfm8hk/detectron2-densepose_ab83aaa5a59c4ff4be17e4db5033ec05
  Resolved https://github.com/facebookresearch/detectron2 to commit c47167e4ac236a36895c294735a908b75f659f96
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/robert/anaconda3/envs/zamba/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-p9zfm8hk/detectron2-densepose_ab83aaa5a59c4ff4be17e4db5033ec05/projects/DensePose/setup.py'"'"'; __file__='"'"'/tmp/pip-install-p9zfm8hk/detectron2-densepose_ab83aaa5a59c4ff4be17e4db5033ec05/projects/DensePose/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-9tevkcff
       cwd: /tmp/pip-install-p9zfm8hk/detectron2-densepose_ab83aaa5a59c4ff4be17e4db5033ec05/projects/DensePose
  Complete output (18 lines):
  Traceback (most recent call last):
    File "/tmp/pip-install-p9zfm8hk/detectron2-densepose_ab83aaa5a59c4ff4be17e4db5033ec05/projects/DensePose/setup.py", line 6, in <module>
      import torch  # noqa: F401
  ModuleNotFoundError: No module named 'torch'

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-p9zfm8hk/detectron2-densepose_ab83aaa5a59c4ff4be17e4db5033ec05/projects/DensePose/setup.py", line 8, in <module>
      raise Exception(
  Exception:
  You must install PyTorch prior to installing DensePose:
  pip install torch

  For more information:
      https://pytorch.org/get-started/locally/

  ----------------------------------------
WARNING: Discarding git+https://github.com/facebookresearch/detectron2@main#subdirectory=projects/DensePose. Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
Collecting fvcore
  Downloading fvcore-0.1.5.post20211023.tar.gz (49 kB)
     |████████████████████████████████| 49 kB 4.2 MB/s
  Preparing metadata (setup.py) ... done
Collecting pandas-path
  Using cached pandas_path-0.3.0-py3-none-any.whl (8.4 kB)
Collecting torchinfo
  Using cached torchinfo-1.5.3-py3-none-any.whl (19 kB)
Collecting opencv-python-headless
  Using cached opencv_python_headless-4.5.4.58-cp38-cp38-manylinux2014_x86_64.whl (47.6 MB)
Collecting pydantic
  Using cached pydantic-1.8.2-cp38-cp38-manylinux2014_x86_64.whl (13.7 MB)
Collecting loguru
  Using cached loguru-0.5.3-py3-none-any.whl (57 kB)
Collecting ffmpeg-python
  Using cached ffmpeg_python-0.2.0-py3-none-any.whl (25 kB)
Collecting pandas>1.2.0
  Using cached pandas-1.3.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.5 MB)
Collecting python-dotenv
  Using cached python_dotenv-0.19.1-py2.py3-none-any.whl (17 kB)
Collecting timm
  Using cached timm-0.4.12-py3-none-any.whl (376 kB)
Collecting future
  Using cached future-0.18.2-py3-none-any.whl
Collecting pytorch-lightning
  Downloading pytorch_lightning-1.5.0-py3-none-any.whl (1.0 MB)
     |████████████████████████████████| 1.0 MB 57.8 MB/s
Collecting thop
  Using cached thop-0.0.31.post2005241907-py3-none-any.whl (8.7 kB)
Collecting tensorboard
  Using cached tensorboard-2.7.0-py3-none-any.whl (5.8 MB)
Collecting torchvision>=0.10.0
  Using cached torchvision-0.11.1-cp38-cp38-manylinux1_x86_64.whl (23.3 MB)
Collecting torch
  Using cached torch-1.10.0-cp38-cp38-manylinux1_x86_64.whl (881.9 MB)
Collecting gitpython
  Using cached GitPython-3.1.24-py3-none-any.whl (180 kB)
Collecting scikit-learn
  Using cached scikit_learn-1.0.1-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (25.9 MB)
Collecting av
  Using cached av-8.0.3-cp38-cp38-manylinux2010_x86_64.whl (38.1 MB)
Collecting appdirs
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting cloudpathlib[s3]
  Using cached cloudpathlib-0.6.2-py3-none-any.whl (49 kB)
Collecting openpyxl
  Using cached openpyxl-3.0.9-py2.py3-none-any.whl (242 kB)
Collecting tqdm
  Using cached tqdm-4.62.3-py2.py3-none-any.whl (76 kB)
Collecting typer[all]
  Using cached typer-0.4.0-py3-none-any.whl (27 kB)
Collecting numpy
  Downloading numpy-1.21.4-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |████████████████████████████████| 15.7 MB 55.6 MB/s
Collecting pytest-mock
  Using cached pytest_mock-3.6.1-py3-none-any.whl (12 kB)
Collecting coverage
  Using cached coverage-6.1.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (216 kB)
Requirement already satisfied: wheel in /home/robert/anaconda3/envs/zamba/lib/python3.8/site-packages (from zamba==2.0.0->-r requirements-dev.txt (line 1)) (0.37.0)
Collecting pytest-coverage
  Using cached pytest_coverage-0.0-py2.py3-none-any.whl (2.0 kB)
Collecting pytest
  Using cached pytest-6.2.5-py3-none-any.whl (280 kB)
ERROR: Could not find a version that satisfies the requirement detectron2-densepose (unavailable) (from zamba[densepose,tests]) (from versions: none)
ERROR: No matching distribution found for detectron2-densepose (unavailable)
make: *** [Makefile:28: requirements] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ requirements:
 ifeq (${CPU_OR_GPU}, gpu)
 	conda install cudatoolkit=11.0.3 cudnn=8.0 -c conda-forge
 endif
-	$(PYTHON_INTERPRETER) -m pip install -U pip setuptools wheel
+	$(PYTHON_INTERPRETER) -m pip install -U pip setuptools wheel torch
 	$(PYTHON_INTERPRETER) -m pip install -r requirements-dev.txt
 
 clean: clean-build clean-pyc clean-test ## remove all build, test, coverage and Python artifacts

